backend/fms_core/services/report.py

```diff
@@ -194,21 +194,19 @@ def get_report(report_name: str, grouped_by: List[str], time_window: TimeWindow,
     for entry in queryset:
         current_row = {key: value for key, value in entry.items() if not key=="time_window"}
         report_by_time_window[entry["time_window"]].append(current_row)
-  
     # Creating header
-    fields = [field.name for field in queryset.model._meta.get_fields()]
     if len(grouped_by) == 0:
-        headers = [{**field, "aggregation": None} for field in MetricField.objects.filter(name__in=fields).values("name", "display_name", "field_order", "data_type")]
+        headers = [{**field, "aggregation": None} for field in MetricField.objects.filter(report__name=report_name).values("name", "display_name", "field_order", "data_type")]
     else:
         field_ordering_dict = {}
-        for field in MetricField.objects.filter(name__in=fields).values("name", "display_name", "field_order", "aggregation", "data_type"):
+        for field in MetricField.objects.filter(report__name=report_name).values("name", "display_name", "field_order", "aggregation", "data_type"):
             field_ordering_dict[field["name"]] = field
         # order for grouping fields
         for i, name in enumerate(grouped_by, start=1):
             field_ordering_dict[name]["field_order"] = i
             headers.append(field_ordering_dict[name])
         # order for value fields
-        aggregate_fields = MetricField.objects.filter(name__in=fields).filter(aggregation__isnull=False).values("name", "display_name", "field_order", "aggregation").order_by("field_order")
+        aggregate_fields = MetricField.objects.filter(report__name=report_name).filter(aggregation__isnull=False).values("name", "display_name", "field_order", "aggregation").order_by("field_order")
```

Comment:
Maybe you can reuse `MetricField.objects.filter(report__name=report_name)` plus `select_related`?